api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+swagger: '2.0'
+info:
+  title: Nakadi Event Bus API Definition
+  description: |
+    -----------
+    Definitions
+    -----------
+    Nakadi at its core aims at being a generic and content-agnostic event broker with a convenient API.
+    In doing this, Nakadi abstracts away, as much as possible, details of the backing messaging
+    infrastructure. The single currently supported messaging infrastructure is Kafka (Kinesis is planned
+    for the future).
+
+    In Nakadi every Event has an EventType, and a **stream** of Events is exposed for each registered
+    EventType.
+
+    An EventType defines properties relevant for the operation of its associated stream, namely:
+
+    * The **schema** of the Event of this EventType. The schema defines the accepted format of Events of an
+    EventType and will be, if so desired, enforced by Nakadi. Usually Nakadi will respect the schema for
+    the EventTypes in accordance to how an owning Application defines them.
+    **Note:** *Currently the specification of the schema must be pushed into Nakadi on EventType creation;
+    in the future, assuming that Applications will expose the schema for its owned resources, Nakadi might
+    support fetching the schema directly from them.*
+    Nakadi might define globally minimal schema compliance requirements (i.e. some fields that must be
```

Comment:
"might define"? Confusing -- it's not clear what is meant by schema here.
Event structure consists of 2 parts:
a) generic part -- common structure valid for all events (of a certain class like BusinessEvent)
The scheme of the generic part is defined in this API.
b) dedicated part -- specific for each event type defining the structure of the specific event payload.
The payload scheme of the dedicated part is defined as part of the EventType definition. 
Here you use "schema" in the sense of "playload schema", do you? I would than call it so.
Nakadi restrictions on the "payload schema" are indicated here -- I would not expect it. Which restrictions?

The payload schema is an explicit event interface contract between producers and consumers. 
And Nakadi enforces payload schema definition, correct?
However, when schema validation is activated we will at least support "compatible extension match" validation mode (and perhaps additionally "exact match" mode), do we? Validation rule config options are not yet defined in this API spec.